src/util/duration.h

```diff
@@ -154,6 +154,15 @@ class Duration : public DurationBase {
         return Duration(nanos);
     }
 
+    static constexpr Duration fromStdDuration(
+            std::chrono::duration<std::int64_t, std::nano> duration) {
+        return Duration::fromNanos(duration.count());
+    }
```

Comment:
I think this should become a template, to allow any durations with any Period

Reply:
That would only really result in the duration_cast being hidden from the caller. In this case I chose the simplest solution that would result in the simplest codegen. So yeah, making this a template would not be hard, but it would essentially result in an implicit conversion based on the period, which I thought we might prefer if it was explicit. The code in the caller wouldn't be very complicated either when the duration_cast is explicit thanks to function template argument deduction...

So I won't object to making the cast implicit, I think its just better if it isn't since it keeps the code a little simpler and possibly more performant (since the casts always result in a multiplication or even worse, division).

Reply:
fyi, the duration cast is not even necessary, and thus the template. I wish it wasn't but it is, so you get your easy of use and I get the simple function signature. https://compiler-explorer.com/z/bbEb16zvs